### PR TITLE
DXCDT-140: "Configuring the Deploy CLI" doc page [4/10]

### PR DESCRIPTION
**Pull request by willvedd**
## ✏️ Changes

The fourth of a ten PR cascade to publish the content from the Deploy CLI's documentation overhaul. This one focuses on the "Configuring the Deploy CLI" page.

## 🔗 References

- [Documentation content outline](https://oktawiki.atlassian.net/wiki/spaces/DXCDT/pages/2632975220/Deploy+CLI+Docs+Content+Outline)
- ["Configuring the Deploy CLI" draft](https://oktawiki.atlassian.net/wiki/spaces/DXCDT/pages/2640646232/Configuring+the+Deploy+CLI)

### CODE REVIEW

File: docs/configuring-the-deploy-cli.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+To **disable** the consumption of environment variables for either the `import` or `export` commands, pass the `--env=false` argument.
+
+#### Examples:
+
+```shell
+# Deploying configuration for YAML formats without a config.json file
+AUTH0_DOMAIN=<YOUR_AUTH0_DOMAIN> \
+AUTH0_CLIENT_ID=<YOUR_CLIENT_ID> \
+AUTH0_CLIENT_SECRET=<YOUR_CLIENT_SECRET> \
+a0deploy import --input_file=local/tenant.yaml
```
Comment (sergiughf) [inline docs/configuring-the-deploy-cli.md:39]:
```suggestion
export AUTH0_DOMAIN=<YOUR_AUTH0_DOMAIN>
export AUTH0_CLIENT_ID=<YOUR_CLIENT_ID> 
export AUTH0_CLIENT_SECRET=<YOUR_CLIENT_SECRET> 

a0deploy import --input_file=local/tenant.yaml
```

File: docs/configuring-the-deploy-cli.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+# Disable environment variable ingestion
+a0deploy export -c=config.json --format=yaml --output_folder=local --env=false
+
+# Non-primitive configuration values
+AUTH0_EXCLUDED='["actions","organizations"]'\
+AUTH0_KEYWORD_REPLACE_MAPPINGS='{"ENVIRONMENT":"dev"}'\
```
Comment (sergiughf) [inline docs/configuring-the-deploy-cli.md:46]:
```suggestion
export AUTH0_EXCLUDED='["actions","organizations"]'\
export AUTH0_KEYWORD_REPLACE_MAPPINGS='{"ENVIRONMENT":"dev"}'\
```

File: docs/configuring-the-deploy-cli.md (hunk 1 above)
Comment (sergiughf) [reply docs/configuring-the-deploy-cli.md:39]: Just a suggestion, your example also works on having them run as a full command, but I personally find this suggestion a bit easier to understand.

**Review by sergiughf — APPROVED** [review]

File: docs/configuring-the-deploy-cli.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+{
+  "AUTH0_DOMAIN": "<YOUR_TENANT_DOMAIN>",
+  "AUTH0_CLIENT_ID": "<YOUR_CLIENT_ID>",
+  "AUTH0_ALLOW_DELETE": false
+}
+```
+
+> ⚠️ **NOTE:** Hard-coding credentials is not recommended, and risks secret leakage should this file ever be committed to a public version control system. Instead, passing credentials as [environment variables](#) is considered best practice.
+
+### Environment variables
```
Comment (Widcket) [inline docs/configuring-the-deploy-cli.md:24]: If there are two different ways to configure the tool, and the first one is an H2 (`##`), should the second one be an H2 as well?

File: docs/configuring-the-deploy-cli.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+```json
+{
+  "AUTH0_EXCLUDED": ["organizations", "connections", "hooks"]
+}
+```
+
+### `AUTH0_KEYWORD_REPLACE_MAPPINGS`
+
+Mapping of specific keywords to facilities dynamic replacement. See also: keyword replacement mapping.
```
Comment (Widcket) [inline docs/configuring-the-deploy-cli.md:86]: > See also: keyword replacement mapping.

This is missing the link

File: docs/configuring-the-deploy-cli.md (hunk 3 above)
Comment (willvedd) [reply docs/configuring-the-deploy-cli.md:24]: Yeah, this is a good point. Thanks!

File: docs/configuring-the-deploy-cli.md (hunk 4 above)
Comment (willvedd) [reply docs/configuring-the-deploy-cli.md:86]: This link is added in a downstream PR for the Keyword Replacement page. But was good to double check!

**Pull request merged**